sdk/go/auto/remote_workspace_test.go

```diff
@@ -15,11 +15,214 @@
+			assert.EqualError(t, err, tc.err)
+		})
+	}
+}
+
+func testRemoteStackGitSource(t *testing.T, fn func(ctx context.Context, stackName string, repo GitRepo,
+	opts ...RemoteWorkspaceOption) (RemoteStack, error)) {
+
+	if os.Getenv("PULUMI_ACCESS_TOKEN") == "" {
+		t.Skipf("Skipping: PULUMI_ACCESS_TOKEN is not set")
+	}
+
+	ctx := context.Background()
+	pName := "go_remote_proj"
+	sName := randomStackName()
+	stackName := FullyQualifiedStackName(pulumiOrg, pName, sName)
+	repo := GitRepo{
+		URL:         remoteTestRepo,
+		Branch:      "refs/heads/master",
+		ProjectPath: "goproj",
+	}
+
+	// initialize
+	s, err := fn(ctx, stackName, repo, RemotePreRunCommands(
+		fmt.Sprintf("pulumi config set bar abc --stack %s", stackName),
+		fmt.Sprintf("pulumi config set --secret buzz secret --stack %s", stackName)))
+	if err != nil {
+		t.Errorf("failed to initialize stack, err: %v", err)
+		t.FailNow()
+	}
+
+	defer func() {
+		// -- pulumi stack rm --
+		err = s.stack.Workspace().RemoveStack(ctx, s.Name())
+		assert.Nil(t, err, "failed to remove stack. Resources have leaked.")
+	}()
+
+	// -- pulumi up --
+	res, err := s.Up(ctx)
+	if err != nil {
+		t.Errorf("up failed, err: %v", err)
+		t.FailNow()
+	}
+
+	assert.Equal(t, 3, len(res.Outputs), "expected two plain outputs")
+	assert.Equal(t, "foo", res.Outputs["exp_static"].Value)
+	assert.False(t, res.Outputs["exp_static"].Secret)
+	assert.Equal(t, "abc", res.Outputs["exp_cfg"].Value)
+	assert.False(t, res.Outputs["exp_cfg"].Secret)
+	assert.Equal(t, "secret", res.Outputs["exp_secret"].Value)
+	assert.True(t, res.Outputs["exp_secret"].Secret)
+	assert.Equal(t, "update", res.Summary.Kind)
+	assert.Equal(t, "succeeded", res.Summary.Result)
+
+	// -- pulumi preview --
+
+	var previewEvents []events.EngineEvent
+	prevCh := make(chan events.EngineEvent)
+	wg := collectEvents(prevCh, &previewEvents)
+	prev, err := s.Preview(ctx, optremotepreview.EventStreams(prevCh))
+	if err != nil {
+		t.Errorf("preview failed, err: %v", err)
+		t.FailNow()
+	}
+	wg.Wait()
+	assert.Equal(t, 1, prev.ChangeSummary[apitype.OpSame])
+	steps := countSteps(previewEvents)
+	assert.Equal(t, 1, steps)
+
+	// -- pulumi refresh --
+
+	ref, err := s.Refresh(ctx)
+
+	if err != nil {
+		t.Errorf("refresh failed, err: %v", err)
+		t.FailNow()
+	}
+	assert.Equal(t, "refresh", ref.Summary.Kind)
+	assert.Equal(t, "succeeded", ref.Summary.Result)
+
+	// -- pulumi destroy --
+
+	dRes, err := s.Destroy(ctx)
+	if err != nil {
+		t.Errorf("destroy failed, err: %v", err)
+		t.FailNow()
+	}
+
+	assert.Equal(t, "destroy", dRes.Summary.Kind)
+	assert.Equal(t, "succeeded", dRes.Summary.Result)
+}
+
+func TestSelectRemoteStackGitSourceErrors(t *testing.T) {
+	t.Parallel()
+	testRemoteStackGitSourceErrors(t, SelectRemoteStackGitSource)
+}
+
+func TestNewRemoteStackGitSourceErrors(t *testing.T) {
+	t.Parallel()
+	testRemoteStackGitSourceErrors(t, NewRemoteStackGitSource)
+}
+
+func TestNewRemoteStackGitSource(t *testing.T) {
```

Comment:
Unfortunately, this test takes ~3 minutes. What can we do to avoid running it in PRs?

Reply:
We'll disable the test by default (requires PULUMI_TEST_DEPLOYMENTS_API to be set). And consider running the test in a cron until we add support for mocking the deployments api.